yt/frontends/exodus_ii/data_structures.py

```diff
@@ -336,7 +336,30 @@ def _read_connectivity(self):
         connectivity = []
         with self._handle.open_ds() as ds:
             for i in range(self.parameters["num_meshes"]):
-                connectivity.append(ds.variables["connect%d" % (i + 1)][:].astype("i8"))
+                var = ds.variables["connect%d" % (i + 1)][:].astype("i8")
+                try:
+                    elem_type = var.elem_type.lower()
+                    if elem_type == "nsided":
+                        arbitrary_polyhedron = True
+                    elif elem_type == "nfaced":
+                        raise NotImplementedError(
+                            "3D arbitrary polyhedra are not implemented yet"
+                        )
+                    else:
+                        arbitrary_polyhedron = False
```

Comment:
I think this won't result in the correct behavior for files that use another `elem_type` (like `quad` etc.) because they would all be set to `nsided`.
As far as I know `nsided` and `nfaced` are the only two types that don't have the same number of nodes for each element, so they need special treatment.

Reply:
I found a comprehensive list of possible values for `elem_type` in the source of the VTK Exodus writer.
https://gitlab.kitware.com/vtk/vtk/-/blob/master/IO/Exodus/vtkExodusIIWriter.cxx#L1200

Reply:
But this doesn't assign `elem_type` !
The last lign assigns the Boolean result of `elem_type == "nsided"` to `arbitrary_polyhedron`, so it's just the compact form of what you wrote. :)

Reply:
Ah, sorry, I missed the double equals. I agree then.